src/module.ts

```diff
@@ -1,14 +1,59 @@
-import { DataSourcePlugin } from '@grafana/data';
+import { DataSourcePlugin, DashboardLoadedEvent } from '@grafana/data';
 import ConfigEditor from 'components/ConfigEditor';
+import { getAppEvents } from '@grafana/runtime';
+import pluginJson from './plugin.json';
 
 import { AdxDataSource } from './datasource';
 import { QueryEditor } from './components/QueryEditor';
-import { AdxDataSourceOptions, AdxDataSourceSecureOptions, KustoQuery } from './types';
+import { AdxDataSourceOptions, AdxDataSourceSecureOptions, FormatOptions, KustoQuery } from './types';
 import EditorHelp from 'components/QueryEditor/EditorHelp';
+import { trackADXMonitorDashboardLoaded } from 'tracking';
 
 export const plugin = new DataSourcePlugin<AdxDataSource, KustoQuery, AdxDataSourceOptions, AdxDataSourceSecureOptions>(
   AdxDataSource
 )
   .setConfigEditor(ConfigEditor)
   .setQueryEditorHelp(EditorHelp)
   .setQueryEditor(QueryEditor);
+
+// Track dashboard loads to RudderStack
+getAppEvents().subscribe<DashboardLoadedEvent<KustoQuery>>(
+  DashboardLoadedEvent,
+  ({ payload: { dashboardId, orgId, grafanaVersion, queries } }) => {
+    const adxQueries = queries[pluginJson.id]?.filter((q) => !q.hide);
+    if (!adxQueries?.length) {
```

Comment:
```suggestion
    if (adxQueries && adxQueries.length > 0) {
```

As the condition around `trackADXMonitorDashboardLoaded` can be removed